compactor_scheduler/src/error.rs

```diff
@@ -0,0 +1,37 @@
+//! Error classification.
+
+/// What kind of error did we occur during compaction?
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub enum ErrorKind {
+    /// Could not access the object store.
+    ObjectStore,
+
+    /// We ran out of memory (OOM).
+    OutOfMemory,
+
+    /// Partition took too long.
+    Timeout,
+
+    /// Unknown/unexpected error.
+    ///
+    /// This will likely mark the affected partition as "skipped" and the compactor will no longer touch it.
+    Unknown,
```

Comment:
```suggestion
    Unknown(String),
```

should we at least transmit some readable string in this case?

Reply:
Done. 

Note that I didn't do the same for the compactor's ErrorKind::Unknown, since it be a larger diff (impacts more logic handling variants and HashSet tracking). Is doable, but it may be better done separately.